apps/nextra/pages/en/build/create-aptos-dapp/templates/token-staking-dapp.mdx

```diff
@@ -0,0 +1,177 @@
+---
+title: "Create Aptos Dapp Token Staking Dapp Template"
+---
+
+import { Callout, FileTree, Steps } from 'nextra/components';
+
+# Create Aptos Dapp Token Staking Dapp Template
+
+The Token staking dapp template provides an end-to-end Fungible Asset staking dapp. With this dapp, you can learn how Fungible Assets work on Aptos, how to write a smart contract and how to write a front end connected to the contract. In addition, it offers a beautiful pre-made UI for a Fungible Asset staking dapp users can quickly adjust and deploy into a live server.
+
+<Callout>
+If you have not created a fungible asset yet, check out the [Token Minting Dapp Template](./token-minting-dapp) to easily create a token on Aptos.
+</Callout>
+
+The Token staking dapp template provides a one page with different components:
+
+1. Create an incetivize pool - A component for the stake creator to create an incentivize (rewards) pool
+2. Stake Fungible Asset Component - A component for anyone to stake the fungible asset
+3. Unstake Fungible Asset Component - A component for any staker to unstake the fungible asset
+4. Claim fungible assets rewards - A component for any staker to claim the fungible asset rewards
+
+
+## Quick Start
+- [Create a stake on chain](#publish-the-move-contract)
+- [Create an incentivize pool](#create-an-incentivize-pool)
+- [Deploy to a live server](#deploy-to-a-live-server)
+
+## Setup
+
+You can start the app by running:
+
+```bash 
+npm run dev
+```
+
+However, for the app to work as intended, there are several setup steps you will have to do to:
+
+1. Publish the contract.
+2. Create an incentivize pool.
+
+## Publish the Move Contract
+
+<Steps>
+
+### On the project root folder, use the below command to initiliaze an account to publish the contract with.
+
+```bash 
+npm run move:init
+```
+
+The CLI will prompt you to provide the private key for your wallet in order for the tool to publish the contract on your behalf. If you provide no input, it will generate one for you.
+
+This will create an account you will later publish the contract with, as well as setting the `VITE_MODULE_ADDRESS` in the `.env` file to set the contract object address.
```

Comment:
```suggestion
This will:
1. Create an account you will later publish the contract with
2. Set the `VITE_MODULE_ADDRESS` in the `.env` file to the contract object address.
```